### ISSUE TITLE
Not handling special 401 errors in phase 2 of XBox auth

### ISSUE THREAD

**Issue by kumquat-ir** [MultiMC/MultiMC5#4025]
### Operating System

Linux

### Description of bug

(Arch Linux, installed from AUR/multimc-bin)

(This is possibly the same bug as #4021, but with much more information)
Logging in with a Microsoft account shows stuck progress bar with message switching between "Logging in as XBox user" and "Logging in with XBox and Mojang Services" after clicking through prompts in browser

Running MultiMC in a terminal shows
```
Second layer of XBox auth ... commencing.
O2Requestor::onRequestError: Error 204
O2Requestor::onRequestError: HTTP status 401 "Unauthorized"
```
repeated many times

After some time, MultiMC crashes with
```
Failed to parse response from user.auth.xboxlive.com as JSON:  "illegal value"
""
Could not parse user authentication response...
terminate called after throwing an instance of 'int'
/usr/bin/multimc: line 23: 275959 Aborted                 (core dumped) ./MultiMC "$@"
```

### Steps to reproduce

* Attempt to log into MultiMC with a Microsoft account that is unable to auth properly (details on that below)

### Suspected cause

(source: https://wiki.vg/Microsoft_Authentication_Scheme#Authenticate_with_XSTS)
This step in MSA authentication can return a 401 error with one of two specific error messages, for accounts that are invalid due to either:
* Not having an XBox account associated
* Under 18 years of age and not part of a family

I suggest to catch and handle these special errors, displaying a message to the user about why auth has failed and cancelling the current auth attempt

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by wntiv-main** [comment]
I'm having this problem too. I can provide full logs if required.

**Comment by thomasz05** [comment]
update my date of birth worked

**Comment by kumquat-ir** [comment]
Fixed in 23442442d86862536acab5a7deca28c1c703b11e
